### PR TITLE
Moving tests related to Pulp2 to a dedicated dir.

### PR DESCRIPTION
**Pull request by kersommoura**
Why?

* Pulp-Smash will be used to test different versions X releases of Pulp.
  Every X release has specific needs from each other. It is a major software change.
  Right now, there are 2 main Pulp X releases: `Pulp2` and `Pulp3`.

* Since both releases will be tested using Pulp-Smash, it is convienent to separte the tests, and files
  related for a specific release. This will help address different needs, and run different tests in a relative
  easy manner.

* To fulfil the the aforementioned needs, it was created a dedicate dir for pulp2, in which one all tests/files related
  to pulp2 were migrated.

* All imports affect by such change were fixed, docstrings were updated, and Makefile were update to reflex the new layout
  adopted for Pulp-Smash. It was created a new `constants.py` inside the pulp2 dir, this file will contain specific constants
  only related to pulp2, mainly constants with the follow standard `pulp/api/v2/{action}`.

* Same approach described above will be adopted to pulp3 dir.

See: #791

### CODE REVIEW

**Comment by kersommoura** [comment]
@Ichimonji10, I ran all tests using `python -m unittest discover pulp_smash.pulp2.tests` against a RHEL7 and F26, with Pulp2.14.

File: pulp_smash/pulp2/tests/puppet/api_v2/test_sync_publish.py [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 
 This module assumes that the tests in
-:mod:`pulp_smash.tests.platform.api_v2.test_repository` hold true. The
+:mod:`pulp_smash.pulp2.tests.platform.api_v2.test_repository` hold true. The
 following trees of assumptions are explored in this module::
 
     It is possible to create an puppet repo with a feed. (CreateTestCase)
     ├── If a valid feed is given, the sync completes and no errors are
-    │   reported for both valid and ivalid query. (SyncValidFeedTestCase)
+    │   reported for both valid and invalid query. (SyncValidFeedTestCase)
```
Comment (Ichimonji10) [inline pulp_smash/pulp2/tests/puppet/api_v2/test_sync_publish.py]: Nice catch. :wink:

**Comment by kersommoura** [comment]
@Ichimonji10, I am using **commonmark** to verify my commit message, it does look ok using their website, but when I look here, it does not look that great. I will try to fix this commit message again.

**Comment by Ichimonji10** [comment]
Don't stress out over making all the various compilers happy. GitHub flavored markdown has some weird quirks. Notably, GitHub treats a single line break as indicating a newline, whereas the original markdown compiler and common mark don't. I'd recommend just satisfying the most standard implementation (CommonMark) and generally ignoring the weirdness of the others.

**Comment by Ichimonji10** [comment]
As per a discussion we had today, let's place Pulp 2 tests in `pulp_smash.tests.pulp2` and Pulp 3 tests in `pulp_smash.tests.pulp3`.

**Comment by kersommoura** [comment]
See discussion: [Layout Pulp-Smash](https://github.com/PulpQE/pulp-smash/issues/791)

**Comment by kersommoura** [comment]
See discussion: [Layout Pulp-Smash](https://github.com/PulpQE/pulp-smash/issues/791)

**Pull request closed without merging**